modules/crud/sql/update.go

```diff
@@ -243,22 +292,66 @@ func flattenForDate(m *map[string]interface{}) error {
 	for k, v := range *m {
 		mm, ok := v.(map[string]interface{})
 		if !ok {
-			return utils.ErrInvalidParams
+			return fmt.Errorf("invalid current date format (%v) provided", reflect.TypeOf(v))
 		}
-		for _, val := range mm {
-			val, ok := val.(string)
+		for _, valTemp := range mm {
+			val, ok := valTemp.(string)
 			if !ok {
-				return utils.ErrInvalidParams
+				return fmt.Errorf("invalid current date type (%v) provided", valTemp)
 			}
 			switch val {
 			case "date":
 				(*m)[k] = "CURRENT_DATE"
 			case "timestamp":
 				(*m)[k] = "CURRENT_TIMESTAMP"
 			default:
-				return utils.ErrInvalidParams
+				return fmt.Errorf("invalid current date value (%s) provided", val)
 			}
 		}
 	}
 	return nil
 }
+
+func (s *SQL) sanitiseUpdateQuery(sqlString string) string {
+	var placeholder byte
+	switch utils.DBType(s.dbType) {
+	case utils.Postgres, utils.SqlServer:
+		placeholder = '$'
+	}
+	log.Println("sqlString", sqlString, "PlaceHolder", placeholder)
+	var start bool
+	for i := 0; i < len(sqlString); i++ {
+		c := sqlString[i]
+		log.Println("c", c, placeholder)
+		if c == placeholder {
+			start = true
+		}
+		if start && (c == ' ' || c == ',') {
+			sqlString = sqlString[0:i] + ")" + sqlString[i:]
+			start = false
+		}
+		if strings.HasPrefix(sqlString[i:], "WHERE") {
+			break
+		}
+	}
+	return sqlString
+}
+func (s *SQL) sanitiseUpdateQuery2(sqlString string) string {
+
+	for i := 0; i < len(sqlString); i++ {
```

Comment:
u can use range statement here

Reply:
Ranging causes iteration over `rune` values which leads to incorrect indexes. We need to iterate over the individual bytes in this case. Discussed with Noorain